homework/index.js

```diff
@@ -30,17 +15,204 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+  function sortElements(repositories) {
+    return repositories.sort((a, b) => a.name.localeCompare(b.name));
+  }
+
+  function getSelectedOption(sel) {
+    let opt;
+    for (let i = 0; i < sel.options.length; i++) {
+      opt = sel.options[i];
+      if (opt.selected === true) {
+        break;
+      }
+    }
+    return opt;
+  }
+
+  function getSelectedIndex(name, opt) {
+    let selectedIndex = 0;
```

Comment:
Hi Ahmet, just like the previous function, there is a possibility to do away with the for loops as JavaScript now provides an inbuilt method to search for a particular element's index. It's called `findIndex`. I think this method provides the same functionality that you wish to achieve with the for loop. Please have a look here for its documentation - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex